1.2/percentile_test.go

```diff
@@ -22,9 +22,9 @@ func TestGeneratePercentile(t *testing.T) {
 	for _, test := range tests {
 		t.Run(fmt.Sprintf("PASS"), func(t *testing.T) {
 			if output, err := generatePercentile(test.input); err != nil {
-				t.Errorf("Got error, exprected no error")
+				t.Errorf("expected no error, got: %v", err)
 			} else if output(data) != test.p {
-				t.Errorf("Fail expected: %v", test.p)
+				t.Errorf("Fail expected: %v, got: %v", test.p, output)
```

Comment:
`output` is a function.  I don't think printing its value is going to be useful, that'll get a pointer address like `0x49b960`.  You'll want to print the value that output returns.